packages/docs/Examples/sale.md

```diff
@@ -3,20 +3,48 @@ order: -40
 icon: flame
 ---
 
-# Sale
+# Sales
 
-In the previous example both smart objects as well as their owners had to be known before the swap transaction could be built. In this section we show how a sale offer can be built by a seller with needing to know the buyer in advance.
+In the case of a swap, both smart objects have to be known before the swap transaction can be built. In this section we show how a single smart object can be offered for sale at a specified price.
 
-## Smart Contract
+## Sale
 
-We will be reusing the `Swap` smart contract from above, but we also need a new smart contract `Payment` below. Buyer will use this contract to create a smart object holding the satoshis to pay for the NFT
+We first explain how to execute the sale of a smart object. The smart contract is quite simple but the swap does not preserving ordinal ranges of the smart object being sold. It is therefore not safe to use with ordinals. In the next section we explain a slightly more complicated swap that can be used with ordinals. 
+
+### Smart Contract
 
 ```ts
-class Payment extends Contract {
-  _owners: string[]
+class Sale extends Contract {
+  static exec(a: NFT, b: NFT) {
+    const [ownerA] = a._owners
+    const [ownerB] = b._owners
+    a.transfer(ownerB)
+    b.transfer(ownerA)
+    return [b, a]
+  }
+}
+```
+
+#### Sighash Types
+
+The idea is to create a transaction with two inputs and two outputs. The inputs spend the smart object `a` to be sold as well as another smart object `b` that is used for the payment. Like in the case of a swap the smart contract exchanges the owners of the smart objects. However unlike the swap smart contract it returns `[b, a]`. This has the effect that the first output of the transaction will represent `b` after the call and the second will represent `a`.
```

Comment:
Can we be more specific on what means that "the first output of the transaction will represent `b` after the call and the second will represent `a`."?